tests/component_tests.py

```diff
@@ -127,3 +135,48 @@ def test_GIVEN_ioc_xml_WHEN_simlevel_is_not_none_THEN_get_ioc_in_sim_mode_return
+        [("LINMOT_{:02d}".format(i), "AXIS", "^AXIS[1-8]$", "^yes$") for i in range(1, 4)] +
+        [("KHLY2001_01", "CHANNEL ACTIVATED", "^ACTIVATE_CHAN_0[1-9]$", "^1$")] +
+        [("NWPRTXPS_01", "AXIS", "^AXIS[1-4]_ID$", "^.*[.].*$")]
+    )
+    @skip_on_instruments(["SELAB", "DEMO"], "Allowed invalid iocs, these are fake instruments")
+    def test_GIVEN_a_component_THEN_for_each_ioc_present_at_least_one_macro_set(self, ioc, macro_name, macro_regex, value_regex):
+        iocs_xml = self.component_utils.get_iocs_xml(self.component)
+        iocs = self.component_utils.get_iocs(iocs_xml)
+        
+        if ioc in iocs:
+            component_macros, globals_macros = self.component_utils.check_ioc_has_macros_with_name(self.component_utils.get_ioc_macros(iocs_xml, ioc), 
+                        self.global_utils.get_macros(ioc), macro_regex)
+            self.assertTrue(len(component_macros)!=0 or len(globals_macros)!=0, 
+                        "No {} macros found in {} in component {}".format(macro_name, ioc, self.component))
+
+            component_macros, globals_macros = self.component_utils.check_ioc_has_macros_with_value(component_macros, 
+                        globals_macros, value_regex)
+            self.assertTrue(len(component_macros) != 0 or len(globals_macros) != 0, 
+                        "At least one {} macro in {} not set in component {}".format(macro_name, ioc, self.component))
+    
+    @parameterized.expand(
+        [("MERCURY_{:02d}".format(i), "TEMPERATURE/LEVEL/PRESSURE", "^(TEMP_[1-4]|LEVEL_[1-2]|PRESSURE_[1-2])$", "^.*[.].*$") 
+            for i in range(1, 3)]
+    )
+    @skip_on_instruments(["SELAB", "DEMO"], "Allowed invalid iocs, these are fake instruments")
+    @skip_on_instruments(["LARMOR", "ZOOM", "IRIS", "SANDALS", "GEM", "MAPS", "OSIRIS", "LET"], "Mercury iTC macros on these instruments are out of date")
+    def test_GIVEN_a_component_THEN_for_each_ioc_present_at_least_one_macro_set(self, ioc, macro_name, macro_regex, value_regex):
+        iocs_xml = self.component_utils.get_iocs_xml(self.component)
+        iocs = self.component_utils.get_iocs(iocs_xml)
+        
+        if ioc in iocs:
+            component_macros, globals_macros = self.component_utils.check_ioc_has_macros_with_name(self.component_utils.get_ioc_macros(iocs_xml, ioc), 
+                        self.global_utils.get_macros(ioc), macro_regex)
+            self.assertTrue(len(component_macros)!=0 or len(globals_macros)!=0, 
+                        "No {} macros found in {} in component {}".format(macro_name, ioc, self.component))
+
+            component_macros, globals_macros = self.component_utils.check_ioc_has_macros_with_value(component_macros, 
+                        globals_macros, value_regex)
+            self.assertTrue(len(component_macros) != 0 or len(globals_macros) != 0, 
+                        "At least one {} macro in {} not set in component {}".format(macro_name, ioc, self.component))
```

Comment:
This test will get picked up as it has a unique name due to the parameterisation, but i think we should still name it something unique like `test_GIVEN_a_component_THEN_for_each_mercury_ioc_present_at_least_one_macro_set`

I also think we could call the original test instead of copying the implementation here?